### PR TITLE
Refactor `feature_gate.rs` into modules & cleanup

### PR DESCRIPTION
**Pull request by Centril**
Split `src/libsyntax/feature_gate.rs` into `src/libsyntax/feature_gate/` with files:
- `accepted.rs` (accepted feature gates)
- `removed.rs` (...)
- `active.rs` (...)
- `builtin_attrs.rs` (definition of builtin attributes and their gates as well as gating `cfg` flags)
- `check.rs` (post expansion checking of feature gates)
- `mod.rs` (just reexports)

Additionally, `tidy.rs` is adjusted to respect the new scheme.

Also, `builtin_attrs.rs` sees some cleanup, organization, and DSL-ification to reduce repetition.

This is probably best read commit-by-commit I think.

r? @oli-obk

### CODE REVIEW

**Review by oli-obk — COMMENTED** [review]
r=me with question resolved

File: src/libsyntax/feature_gate.rs [hunk 1]
```diff
@@ -42,523 +45,6 @@ use lazy_static::lazy_static;
...
-            pub fn walk_feature_fields<F>(&self, mut f: F)
-                where F: FnMut(&str, bool)
-            {
-                $(f(stringify!($feature), self.$feature);)+
-            }
-        }
-    };
-}
-
-// If you change this, please modify `src/doc/unstable-book` as well.
```
[the remark this answers is not in the document]
Comment (Centril) [reply src/libsyntax/feature_gate.rs]: The comment is moved over to `active.rs` which seemed like the most appropriate place to have it. The latter bits are still relevant but I don't know how much the unstable book still is but that seems like a larger question which is preferably resolved in a more dedicated place. The file paths are still right however.

[the remark this answers is not in the document]
Comment (Centril) [reply src/libsyntax/feature_gate.rs]: Filed as an issue in https://github.com/rust-lang/rust/issues/63840.

**Pull request merged**